### PR TITLE
[DO NOT MERGE] Global search PR

### PR DESCRIPTION
**Pull request by smathangi**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -18,8 +18,9 @@ def component = "data-store-api"
 def branchesToSync = ['demo', 'ithc', 'perftest']
 
 // Variables to switch pipeline logic and wiring per type of build
-def definitionStoreDevelopPr = "PR-575"  // This doesn't change frequently, but when it does, only change this value.
-def dataStoreApiDevelopPr    = "PR-1260" // This doesn't change frequently, but when it does, only change this value.
+
+def definitionStoreDevelopPr = "PR-1073" // This doesn't change frequently, but when it does, only change this value.
+def dataStoreApiDevelopPr    = "PR-1652" // This doesn't change frequently, but when it does, only change this value.
```
Comment (mattnayler) [inline Jenkinsfile_CNP:23]: These values should be restored priro to merger.

File: Jenkinsfile_CNP [hunk 2]
```diff
@@ -92,6 +93,7 @@ env.PACT_BROKER_FULL_URL = 'https://pact-broker.platform.hmcts.net'
 env.DEFAULT_COLLECTION_ASSERTION_MODE="UNORDERED"
 env.MAX_NUM_PARALLEL_THREADS=6
 env.BEFTA_S2S_CLIENT_ID = "ccd_gw"
+env.DEFINITION_STORE_HOST= "http://ccd-definition-store-api-pr-1073.service.core-compute-preview.internal"
```
Comment (mattnayler) [inline Jenkinsfile_CNP:96]: These values should be restored priro to merger.

File: charts/ccd-data-store-api/values.preview.template.yaml [hunk 3]
```diff
@@ -23,7 +23,7 @@ java:
     DATA_STORE_DB_PASSWORD: "{{ .Values.postgresql.postgresqlPassword}}"
     DATA_STORE_DB_OPTIONS: "?stringtype=unspecified"
     DATA_STORE_DB_MAX_POOL_SIZE: 10
-    DEFINITION_STORE_HOST: http://ccd-definition-store-api-pr-575.service.core-compute-preview.internal/
+    DEFINITION_STORE_HOST: http://ccd-definition-store-api-pr-1073.service.core-compute-preview.internal/
```
Comment (mattnayler) [inline charts/ccd-data-store-api/values.preview.template.yaml:26]: These values should be restored priror to merger.

File: build.gradle [hunk 4]
```diff
@@ -581,7 +581,7 @@ jsonSchema2Pojo {
 
 sonarqube {
     properties {
-        property "sonar.exclusions", "build/generated-sources/**/*.java,**/AuditRepository.java,**/CaseDetailsEndpoint.java,**/MidEventCallback.java,**/UIDefinitionRepository.java,**/CachedUIDefinitionGateway.java,**/CaseAccessEndpoint.java,**/UICaseSearchController.java,**/CaseSearchResultViewGenerator.java,**/CaseTypeService.java,**/DefaultCaseDefinitionRepository.java"
+        property "sonar.exclusions", "build/generated-sources/**/*.java,**/AuditRepository.java,**/CaseDetailsEndpoint.java,**/MidEventCallback.java,**/UIDefinitionRepository.java,**/CachedUIDefinitionGateway.java,**/CaseAccessEndpoint.java,**/UICaseSearchController.java,**/CaseSearchResultViewGenerator.java,**/CaseTypeService.java,**/DefaultCaseDefinitionRepository.java, **/SearchCriteriaResponseTEMP.java"
```
Comment (mattnayler) [inline build.gradle:584]: Is new excluson required?

File: build.gradle [hunk 5]
```diff
@@ -265,7 +265,7 @@ dependencies {
     compile 'org.jooq:jool-java-8:0.9.14'
     compile 'com.github.hmcts:ccd-case-document-am-client:1.7.1'
 
-    testCompile group: 'com.github.hmcts', name: 'ccd-test-definitions', version: '7.8.1'
+    testCompile group: 'com.github.hmcts', name: 'ccd-test-definitions', version: '7.8.4-Prerelease'
```
Comment (mattnayler) [inline build.gradle:268]: This will need updating when the corresponding ccd-test-definitions updates are merged.

**Review by danlysiak — APPROVED** [review]

File: build.gradle (hunk 5 above)
Comment (MSancaktutar) [reply build.gradle:268]: 7.8.5-pre in def store and 7.8.4-pre here are consistent?
Would be good to use a single latest stable release version in both.

**Review by MSancaktutar — APPROVED** [review]
lgtm. Ditto for comments from @mattnayler , which I think are relatively minor ones.

File: src/main/java/uk/gov/hmcts/ccd/data/definition/CachedCaseDefinitionRepository.java [hunk 6]
```diff
@@ -109,7 +110,7 @@ public CaseTypeDefinitionVersion getLatestVersion(String caseTypeId) {
     @Override
     public JurisdictionDefinition getJurisdiction(String jurisdictionId) {
         LOGGER.debug("Will get jurisdiction '{}' from repository.", jurisdictionId);
-        return caseDefinitionRepository.getJurisdiction(jurisdictionId);
+        return jurisdictions.computeIfAbsent(jurisdictionId, caseDefinitionRepository::getJurisdiction);
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/ccd/data/definition/CachedCaseDefinitionRepository.java:113]: This was meant to have been reverted as part of this PR

**Review by danlysiak — CHANGES_REQUESTED** [review]

**Review by mattnayler — APPROVED** [review]
approved: subect to the minor changes raised as comments. see `Jenkinsfile_CNP` & `build.gradle`

File: Jenkinsfile_CNP [hunk 7]
```diff
@@ -92,6 +93,7 @@ env.PACT_BROKER_FULL_URL = 'https://pact-broker.platform.hmcts.net'
 env.DEFAULT_COLLECTION_ASSERTION_MODE="UNORDERED"
 env.MAX_NUM_PARALLEL_THREADS=6
 env.BEFTA_S2S_CLIENT_ID = "ccd_gw"
+env.DEFINITION_STORE_HOST = "http://ccd-definition-store-api-pr-1073.service.core-compute-preview.internal"
```
Comment (mattnayler) [inline Jenkinsfile_CNP:96]: It feels like this should also be removed priror to merger (?)

**Pull request merged**